extension-iap/src/iap_private.cpp

```diff
@@ -121,13 +121,14 @@ void IAP_Queue_Push(IAPCommandQueue* queue, IAPCommand* cmd)
 void IAP_Queue_Flush(IAPCommandQueue* queue, IAPCommandFn fn, void* ctx)
 {
     assert(fn != 0);
+
+    DM_MUTEX_SCOPED_LOCK(queue->m_Mutex);
+
     if (queue->m_Commands.Empty())
```

Comment:
There is a small window of opportunity when the m_Commands is being reallocated and the internal pointers are being assigned. During this time, it's not safe to compare the internal pointers.

However, I'm not entirely sure if it's actually causing any real troubles in this case, since the mutex should protect the loop itself.

But still, I think the fix is demonstative if nothing else.

Reply:
Yes, it makes sense to move the lock to earlier in the code!